rule_manager.py

```diff
@@ -0,0 +1,135 @@
+
+    def get_rule_by_id(self, rule_id: str) -> Optional[Rule]:
+        """Retrieve a rule by its ID."""
+        return self.rules.get(rule_id)
+
+    def override(self, rule_id: str, **kwargs):
+        """Apply overrides to a specific rule."""
+        if rule_id in self.rules:
+            self.rules[rule_id].override(**kwargs)
+        else:
+            print(f"Warning: Rule with id {rule_id} not found.")
+
+    def apply_overrides(self, overrides_module):
+        overrides_module.apply_overrides(self)
+
+    def set_title(self, rule_id: str, method: Any):
+        self.set_method(rule_id=rule_id, method_name="title", value=method)
+
+    def set_method(self, rule_id: str, method_name: str, value: Any):
+        """
+        Set a dynamic method for a specific rule (e.g., title, severity, destinations).
+
+        Args:
+            rule_id (str): The ID of the rule to modify.
+            method_name (str): The name of the method to set (e.g., 'title', 'severity', 'destinations').
+            value (Any): The value to set for the method. Can be a function or a static value.
+
+        Raises:
+            ValueError: If the rule is not found or the method doesn't exist.
+        """
+        if rule_id not in self.rules:
+            raise ValueError(f"Rule with ID '{rule_id}' not found.")
+
+        rule = self.rules[rule_id]
+
+        if not hasattr(rule, method_name):
+            raise ValueError(f"Method '{method_name}' does not exist for rule '{rule_id}'.")
+
+        # If value is callable, set it directly as the method
+        if callable(value):
+            setattr(rule, method_name, value)
+        else:
+            # If value is not callable, create a lambda function that returns the value
+            setattr(rule, method_name, lambda event: value)
+
+    def include(self, rule_id: str, filter_func):
+        """Apply an include filter to a specific rule."""
+        if rule_id in self.rules:
+            include(filter_func)(self.rules[rule_id])
+        else:
+            print(f"Warning: Rule with id {rule_id} not found.")
+
+    def exclude(self, rule_id: str, filter_func):
+        """Apply an exclude filter to a specific rule."""
+        if rule_id in self.rules:
+            exclude(filter_func)(self.rules[rule_id])
+        else:
+            print(f"Warning: Rule with id {rule_id} not found.")
+
+    def include_bulk(self, log_type: LogType, filter_func):
+        """Apply a filter to all rules of a specific log type."""
+        for rule in self.rules.values():
+            if log_type in rule.log_types:
+                include(filter_func)(rule)
+
+    def exclude_bulk(self, log_type: LogType, filter_func):
+        """Exclude events based on a filter for all rules of a specific log type."""
+        for rule in self.rules.values():
+            if log_type in rule.log_types:
+                exclude(filter_func)(rule)
+
+    def set_property(self, rule_id: str, property_name: str, value: Any):
```

Comment:
Sometimes (like here) an error is raised, sometimes a warning is logged. It seems intentional but I'm not sure which case is done in which case, is one like for internal calls vs. user facing calls?